src/lib/builders/slider/index.ts

```diff
@@ -0,0 +1,252 @@
+import {
+	effect,
+	elementDerived,
+	elementMultiDerived,
+	getElementByMeltId,
+	isBrowser,
+	kbd,
+} from '$lib/internal/helpers';
+import { derived, get, writable } from 'svelte/store';
+
+type CreateSliderArgs = {
+	value: number[];
+	min?: number;
+	max?: number;
+	step?: number;
+	orientation?: 'horizontal' | 'vertical';
+	disabled?: boolean;
+};
+
+const defaults = {
+	value: [],
+	min: 0,
+	max: 100,
+	step: 1,
+	orientation: 'horizontal',
+	disabled: false,
+} satisfies CreateSliderArgs;
+
+export const createSlider = (args: CreateSliderArgs = defaults) => {
+	const withDefaults = { ...defaults, ...args };
+	const value = writable(withDefaults.value);
+	const max = writable(withDefaults.max);
+	const min = writable(withDefaults.min);
+	const disabled = writable(withDefaults.disabled);
+
+	const isActive = writable(false);
+	const currentThumbIndex = writable<number>(0);
+	const activeThumb = writable<{ thumb: HTMLElement; index: number } | null>(null);
+
+	const root = elementDerived(disabled, ($disabled) => {
+		return { disabled: $disabled, 'data-orientation': withDefaults.orientation };
+	});
+
+	const range = derived(value, ($value) => {
+		if (withDefaults.orientation === 'horizontal') {
+			return {
+				style: `position: absolute;
+								left: ${$value.length > 1 ? Math.min(...$value) ?? 0 : 0}%;
+								right: calc(${100 - (Math.max(...$value) ?? 0)}%)`,
+			};
+		} else {
+			return {
+				style: `position: absolute;
+								top: ${$value.length > 1 ? Math.min(...$value) ?? 0 : 0}%;
+								bottom: calc(${100 - (Math.max(...$value) ?? 0)}%)`,
+			};
+		}
+	});
+
+	const getAllThumbs = () => {
+		const rootEl = getElementByMeltId(get(root)['data-melt-id']) as HTMLElement;
```

Comment:
Rather than using get, you can make the other stores have root as one of the stores that it derives from, than pass in the root store to this function. This is preferrable because get can be inneficient performance wise.